dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/Converter.mo

```diff
@@ -0,0 +1,114 @@
+within Dynawo.Electrical.Sources;
+
+/*
+* Copyright (c) 2015-2019, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+*/
+
+model Converter "Converter Model for Grid Forming and Grid Following applications"
+
+/*
+  Equivalent circuit and conventions:
+                       __________
+IdcSourcePu     IdcPu |          |iConvPu                           iPccPu
+-------->-------->----|          |-->-----(Rfilter,Lfilter)---------->--(Rtransformer,Ltransformer)---(terminal)
+              |       |          |                                |
+VdcSourcePu (Cdc)     |  DC/AC   |  uConvPu         uFilterPu (Cfilter)                      uPccPu
+              |       |          |                                |
+              |       |          |                                |
+----------------------|__________|---------------------------------------------------------------------
+
+*/
+
+
+  import Modelica.Math;
+  import Modelica.ComplexMath;
+  import Dynawo.Types;
+  import Dynawo.Connectors;
+  import Dynawo.Electrical.SystemBase;
+
+  Connectors.ImPin IdcSourcePu(value(start = IdcSource0Pu)) "DC Current generated by the DC current source in p.u (base UNom, SNom)";
+  Connectors.ImPin udConvRefPu(value(start = UdConv0Pu)) "d-axis modulated voltage reference in p.u (base UNom)";
```

Comment:
Did you try to use RealInput instead of ImPin? It works fine for InjectorBG for the SVarC which is similar...